File: ResumeSite/resumesite.client/src/Models/Analytics.ts
export interface AnalyticsModel {
    visitorsPerRegions: VisitorsPerRegion[];
    visitorsPerBrowsers: VisitorsPerBroswer[];
    visitorsPerOperatingSystems: VisitorsPerOperatingSystem[];
    visitorsPerDeviceTypes: VisitorsPerDeviceType[];
    visitorsPerCountries: VisitorsPerCountry[];
    visitorsPerCities: VisitorsPerCity[];
    visitorsPerWeathers: VisitorsPerWeather[];
    visitorsPerMonths: VisitorsPerMonth[];
}

export interface VisitorsPerRegion {
    regionName: string;
    count: number;
}

export interface VisitorsPerBroswer {
    browser: string;
    count: number;
}

export interface VisitorsPerOperatingSystem {
    operatingSystem: string;
    count: number;
}

export interface VisitorsPerDeviceType {
    deviceType: string;
    count: number;
}

export interface VisitorsPerCountry {
    country: string;
    count: number;
}

export interface VisitorsPerCity {
    city: string;
    count: number;
}

export interface VisitorsPerWeather {
    weatherCondition: string;
    count: number;
}

export interface VisitorsPerMonth {
    month: string;
    count: number;
}

export interface ChartDataItem {
  label: string;
  value: number;
}

File: ResumeSite/ResumeSite.Server/DTO/Analytics.cs
namespace ResumeSite.Server.DTO
{
    public class Analytics
    {
        public List<VisitorsPerRegion> VisitorsPerRegions { get; set; } = new List<VisitorsPerRegion>();
        public List<VisitorsPerBroswer> VisitorsPerBrowsers { get; set; } = new List<VisitorsPerBroswer>();
        public List<VisitorsPerOperatingSystem> VisitorsPerOperatingSystems { get; set; } = new List<VisitorsPerOperatingSystem>();
        public List<VisitorsPerDeviceType> VisitorsPerDeviceTypes { get; set; } = new List<VisitorsPerDeviceType>();
        public List<VisitorsPerCountry> VisitorsPerCountries { get; set; } = new List<VisitorsPerCountry>();
        public List<VisitorsPerCity> VisitorsPerCities { get; set; } = new List<VisitorsPerCity>();
        public List<VisitorsPerWeather> VisitorsPerWeathers { get; set; } = new List<VisitorsPerWeather>();
        public List<VisitorsPerMonth> VisitorsPerMonths { get; set; } = new List<VisitorsPerMonth>();
    }

    public class VisitorsPerRegion
    {
        public string RegionName { get; set; } = string.Empty;
        public int Count { get; set; } = 0;
    }
    public class VisitorsPerBroswer
    {
        public string Browser { get; set; } = string.Empty;
        public int Count { get; set; } = 0;
    }
    public class VisitorsPerOperatingSystem
    {
        public string OperatingSystem { get; set; } = string.Empty;
        public int Count { get; set; } = 0;
    }
    public class VisitorsPerDeviceType
    {
        public string DeviceType { get; set; } = string.Empty;
        public int Count { get; set; } = 0;
    }
    public class VisitorsPerCountry
    {
        public string Country { get; set; } = string.Empty;
        public int Count { get; set; } = 0;
    }
    public class VisitorsPerCity
    {
        public string City { get; set; } = string.Empty;
        public int Count { get; set; } = 0;
    }
    public class VisitorsPerWeather
    {
        public string WeatherCondition { get; set; } = string.Empty;
        public int Count { get; set; } = 0;
    }
    public class VisitorsPerMonth
    {
        public DateTime Month { get; set; } = DateTime.UtcNow;
        public int Count { get; set; } = 0;
    }
}

